File: c/dicerollgame.c
#include <ctype.h>
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int SumofThreeDice (int *i, int *diceRoll, int *sum)
{
    srand(time(0));
    for (*i = 0; *i < 3; (*i)++)
    {
        *diceRoll = (rand() % 6) + 1;
        *sum += *diceRoll;
    }
    return *sum;
}

int main()
{
    int sum1 = 0, sum2 = 0, i, diceRoll;
    char guess;
    printf("\t-----GUESS THE DICE-----\n\n");
    do
    {
        printf("Press enter to roll three dice");
    } while (getchar() != '\n');
    sum1 = SumofThreeDice(&i, &diceRoll, &sum1);
    printf("The sum of three dice is %d\n", sum1);
    printf("Now, guess the next roll, is it going to be higher, lower, or same!\n(Input l for lower, h for higher, s for same)\n");
    do
    {
        scanf("%c", &guess);
    } while (tolower(guess) != 'l' && tolower(guess) != 'h' && tolower(guess) != 's');
    sum2 = SumofThreeDice(&i, &diceRoll, &sum2);
    printf("Next roll, the sum of three dice is %d\n", sum2);
    if (tolower(guess) == 'l')
    {
        if (sum1 > sum2)
        {
            printf("Good job ! :)");
        }
        else
        {
            printf("You suck ! :(");
        }
    }
    if (tolower(guess) == 'h')
    {
        if (sum1 < sum2)
        {
            printf("Good job ! :)");
        }
        else
        {
            printf("You suck ! :(");
        }
    }
    if (tolower(guess) == 's')
    {
        if (sum1 == sum2)
        {
            printf("Good job ! :)");
        }
        else
        {
            printf("You suck ! :(");
        }
    }
}

File: python/dicerollgame.py
import random

def SumofThreeDice():
    total = 0
    for i in range(3):
        total += random.randint(1, 6)
    return total

def CheckYourGuess(guess, sum1, sum2):
    if guess == 'l':
        if sum1 > sum2:
            greet = 'Good Job ! :)'
        else:
            greet = 'You suck ! :('
    if guess == 'h':
        if sum1 < sum2:
            greet = 'Good Job ! :)'
        else:
            greet = 'You suck ! :('
    if guess == 's':
        if sum1 == sum2:
            greet = 'Good Job ! :)'
        else:
            greet = 'You suck ! :('
    return greet

print('\t-----GUESS THE DICE-----\n')
while True:
    while True:
        usrinput = input('Press enter to roll three dice')
        if usrinput == '':
            break
    sum1 = SumofThreeDice()
    print(f'The sum of three dice is {sum1}')
    print('''Now, guess the next roll, is it going to be higher, lower or same!
(Input l for lower, h for higher, s for same)''')
    while True:
        guess = input()
        guess = guess.lower()
        if guess == 'l' or guess == 'h' or guess == 's':
            break
    sum2 = SumofThreeDice()
    print(f'Next roll, the sum of three dice is {sum2}')
    print(CheckYourGuess(guess, sum1, sum2))
    while True:
        playagain = input('Do you want to play again(y/n) : ')
        playagain = playagain.lower()
        if playagain == 'y' or playagain == 'n':
            break
    if playagain == 'n':
        print('\t-----THANKS FOR PLAYING-----')
        break
